platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/ReconnectLearner.java

```diff
@@ -156,6 +156,7 @@ public ReservedSignedState execute(@NonNull final SignedStateValidator validator
             receiveSignatures();
             final ReservedSignedState reservedSignedState = reconnect();
             validator.validate(reservedSignedState.get(), addressBook, stateValidationData);
+            ReconnectUtils.endReconnectHandshake(connection);
```

Comment:
Could this handshake take place _before_ validating the received state?

Reply:
The whole idea of the handshake is to tell our peer that we are done with all the work. If we placed it before validation, and validation takes a long time, we have the same issue we are trying to solve.